dhis-2/dhis-api/src/main/java/org/hisp/dhis/datavalue/DataValue.java

```diff
@@ -281,24 +282,77 @@ public void mergeWith(DataValue other) {
    * @param newDataElement {@link DataElement} to be used in the new {@link DataValue}
    * @return new {@link DataValue}
    */
-  public static DataValue dataValueWithNewDataElement(DataValue oldDv, DataElement newDataElement) {
-    DataValue newValue =
-        DataValue.builder()
-            .dataElement(newDataElement)
-            .period(oldDv.getPeriod())
-            .source(oldDv.getSource())
-            .categoryOptionCombo(oldDv.getCategoryOptionCombo())
-            .attributeOptionCombo(oldDv.getAttributeOptionCombo())
-            .value(oldDv.getValue())
-            .storedBy(oldDv.getStoredBy())
-            .lastUpdated(oldDv.getLastUpdated())
-            .comment(oldDv.getComment())
-            .followup(oldDv.isFollowup())
-            .deleted(oldDv.isDeleted())
-            .build();
-    newValue.setCreated(oldDv.getCreated());
-    return newValue;
-  }
+  public static final BiFunction<DataValue, BaseIdentifiableObject, DataValue>
```

Comment:
There is a benefit of having this code close the model object to handle new/updated properties. Though this code may be a bit massive to keep on the model object itself? Perhaps move to another class?

Reply:
Yep, can move to a util class 👍